rfcs/0004-page-transitions.md

```diff
@@ -0,0 +1,207 @@
+- Start Date: 2020-01-13
+- Project: [next-with-moxy](https://github.com/moxystudio/next-with-moxy)
+- Reference Issues: N/A
+- Implementation PR: N/A
+
+# Summary
+
+This proposal is to add full page transitions to the `next-with-moxy` boilerplate, either as a guide and easy to use feature (recipe) or an out-of-the-box feature (by including it in the source code).
+
+# Basic example
+
+We will use [Highway](https://highway.js.org/) by Dogstudio, a lightweight library to create the navigation animations.
+
+## Installation
+
+First, install the package:
+
+```sh
+$ npm install --save @dogstudio/highway
+```
+
+## Enabling Highway
+
+In our [`App.js`](https://github.com/moxystudio/next-with-moxy/blob/master/www/app/App.js) file, we'll have to enable Highway:
+
+```js
+import Highway from '@dogstudio/highway/src/highway';
+import { Slide, Fade } from '../shared/components/transitions';
+
+class App extends NextApp {
+    Highway;
+
+    componentDidMount() {
+        this.unregisterGoogleTracking = registerGoogleTracking(this.props.router);
+
+        this.Highway = new Highway.Core({
+            transitions: {
+                '/contact': Fade,
+                default: Slide,
+            },
+        });
+    }
+```
+
+You can learn more about it [here](https://highway.js.org/get-started.html#core), but the gist of it is that the `transitions` object tells which [Transition component](https://highway.js.org/get-started.html#transitions) to use in which route. In this example, you can see that the `/contact` route will use the Fade component and every other route (default) will use the Slide one. We will get to it [in a moment](#transition-component).
+
+## HTML Structure
+
+Highway requires a [`data-router-wrapper`](https://highway.js.org/get-started.html#html-structure) to look inside for a [`data-router-view`](https://highway.js.org/get-started.html#html-structure) to update the content. So, still in our [`App.js`](https://github.com/moxystudio/next-with-moxy/blob/master/www/app/App.js) file:
+
+```js
+render() {
+    const { Component, pageProps, router } = this.props;
+
+    return (
+        <>
+            <Head>
+                {/* ... */}
+            </Head>
+            <KeyboardOnlyOutlines>
+                <div data-router-wrapper>
```

Comment:
Is there any way for us to distinguish the number of wrappers needed?

Reply:
Not that I know of.